src/relay/analysis/get_calibration_data.cc

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file src/relay/analysis/get_calibration_data.cc
+ *
+ * \brief To get the calibration data, we need to perform two
+ * steps. First, we need to prepare the module that generate
+ * the tensor values (GetCalibrateModule). Second, we need to
+ * generate the mapping between the values and the subgraphs
+ * (GetCalibrateOutputMap).
+ */
+
+#include <tvm/relay/expr.h>
+#include <tvm/relay/expr_functor.h>
+#include <tvm/relay/analysis.h>
+
+namespace tvm {
+namespace relay {
+
+/*!
+ * \brief This function returns a module that will be used by
+ * the relay graph runtime for collecting the calibration data.
+ * To do that, we first make all inputs and outputs of each
+ * subgrpah into the final output (i.e., the final output is a
+ * tuple of tensors). Then, we change the compiler attribute of
+ * each subgraph. Finally, we mark all subgraph to be inlined.
+ */
+
+IRModule GetCalibrateModule(IRModule module) {
+  class OutputCollector : public ExprRewriter {
+   public:
+    OutputCollector(const Map<GlobalVar, BaseFunc>& glob_funcs)
+      : glob_funcs(glob_funcs) {}
+
+    Expr Rewrite_(const CallNode* call, const Expr& post) final {
+      if (call->op->IsInstance<GlobalVarNode>()) {
+        auto var = Downcast<GlobalVar>(call->op);
+        // check if it is a subgraph of the original graph
+        if (glob_funcs.count(var) > 0) {
```

Comment:
1. Better to be "check if its function implementation is available".
2. It seems to me that we can still collect its data even if its implementation is unavailable. Although it looks useless for BYOC, it might be useful for other use cases. Maybe we can add an option like `include_intransic_func=False` to the API.